model/src/pyrenew/latent/infection_seeding_method.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABCMeta, abstractmethod
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+from pyrenew.metaclass import RandomVariable
+
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+"""
+Classes for the infection seeding methods
+"""
+
+
+class InfectionSeedMethod(metaclass=ABCMeta):
+    def __init__(self, n_timepoints: int):
+        self.validate(n_timepoints)
+        self.n_timepoints = n_timepoints
+
+    @staticmethod
+    def validate(n_timepoints: int) -> None:
+        if not isinstance(n_timepoints, int):
+            raise TypeError("n_timepoints must be an integer")
+        if n_timepoints <= 0:
+            raise ValueError("n_timepoints must be positive")
+
+    @abstractmethod
+    def seed_infections(self, I0: ArrayLike) -> ArrayLike:
+        pass
+
+
+class SeedInfectionsZeroPad(InfectionSeedMethod):
+    def seed_infections(self, I0: ArrayLike):
+        # alternative implementation:
+        # return jnp.hstack([jnp.zeros(self.n_timepoints - I0.size), I0])
+        return jnp.pad(I0, (self.n_timepoints - I0.size, 0))
```

Comment:
In this class, `I0` has a different meaning than I generally expect it to have. Usually `I0` would be the first element of the array. In this example, `I0` is the last element.

Reply:
This comes back to this question about nomenclature, I think: https://github.com/CDCgov/multisignal-epi-inference/issues/116#issuecomment-2113092808